scipy/stats/morestats.py

```diff
@@ -31,24 +29,4 @@ def __dir__():
 
 
 def __getattr__(name):
-    if name not in __all__:
-        raise AttributeError(
-            f"`scipy.stats.morestats` has no attribute `{name}`; furthermore, "
-            "`scipy.stats.morestats` is deprecated and will be removed in "
-            "SciPy 2.0.0.")
-
-    attr = getattr(scipy.stats, name, None)
-
-    if attr is not None:
-        message = (f"Please import `{name}` from the `scipy.stats` namespace; "
-                   "the `scipy.stats.morestats` namespace is deprecated and "
-                   "will be removed in SciPy 2.0.0.")
-    else:
-        message = (f"`scipy.stats.morestats.{name}` is deprecated along with "
-                   "the `scipy.stats.morestats` namespace. "
-                   f"`scipy.stats.morestats.{name}` will be removed in SciPy 1.13.0, and "
-                   "the `scipy.stats.morestats` namespace will be removed in SciPy 2.0.0.")
-
-    warnings.warn(message, category=DeprecationWarning, stacklevel=2)
-
-    return getattr(_morestats, name)
+    return _sub_module_deprecation("morestats", "stats", "_morestats", __all__, name)
```

Comment:
I think this would be easier to read with named arguments.
```suggestion
    return _sub_module_deprecation(module="stats", submodule="morestats", private_module="_morestats", all=__all__, name=name)
```

Reply:
```suggestion
    return _sub_module_deprecation(subpackage="stats", module="morestats", private_module="_morestats", all=__all__, attribute=name)
```